pinot-core/src/main/java/com/linkedin/pinot/core/operator/query/AggregationOperator.java

```diff
@@ -36,20 +38,23 @@
   private final TransformExpressionOperator _transformOperator;
   private final long _numTotalRawDocs;
   private ExecutionStatistics _executionStatistics;
+  private final Map<String, FieldSpec.DataType> _dataTypeMap;
 
   public AggregationOperator(@Nonnull AggregationFunctionContext[] aggregationFunctionContexts,
-      @Nonnull TransformExpressionOperator transformOperator, long numTotalRawDocs) {
+      @Nonnull TransformExpressionOperator transformOperator, long numTotalRawDocs,
```

Comment:
It might be cleaner (though slightly less efficient) to have the dataType embedded inside aggregationFunctionContexts.